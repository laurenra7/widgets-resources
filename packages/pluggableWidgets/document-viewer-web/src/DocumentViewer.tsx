import { ReactNode, createElement } from "react";
import { DocumentViewer as DocumentViewerComponent } from "./components/DocumentViewer";
import { DocumentViewerContainerProps } from "../typings/DocumentViewerProps";
import { useFileURL } from "./utils/useFileURL";
import { useStyle } from "./utils/useStyle";
import "./ui/DocumentViewer.scss";
import { Alert } from "@mendix/piw-utils-internal/components/web";

export function DocumentViewer(props: DocumentViewerContainerProps): ReactNode {
    const result = useFileURL(props);
    const style = useStyle(props);

    if (result.loading) {
        return null;
    }

    if (result.error !== null) {
        return <Alert bootstrapStyle="danger">{result.error}</Alert>;
    }

    return (
        <DocumentViewerComponent
            className={props.class}
            src={result.data}
            tabIndex={props.tabIndex}
            rootStyle={style}
        />
    );
}
