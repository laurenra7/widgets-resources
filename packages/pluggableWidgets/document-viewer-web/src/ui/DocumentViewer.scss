.widget-document-viewer {
    > .widget-document-viewer-content {
        width: 100%;
        height: 100%;
    }
}
