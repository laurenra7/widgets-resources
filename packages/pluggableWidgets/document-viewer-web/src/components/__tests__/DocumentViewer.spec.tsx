import { createElement } from "react";
import { shallow } from "enzyme";
import { DocumentViewer } from "../DocumentViewer";

describe("DocumentViewer", () => {
    it("renders correctly the structure", () => {
        const viewer = shallow(<DocumentViewer src="http://localhost/sample.pdf" />);
        expect(viewer).toMatchSnapshot();
    });
});
