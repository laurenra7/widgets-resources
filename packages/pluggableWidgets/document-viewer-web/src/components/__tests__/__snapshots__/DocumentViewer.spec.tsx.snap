// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentViewer renders correctly the structure 1`] = `
<div
  className="widget-document-viewer"
>
  <object
    className="widget-document-viewer-content"
    data="http://localhost/sample.pdf"
    type="application/pdf"
  >
    <p>
      We having issues with preview of the current file. Please download the file to view it:Â 
      <a
        href="http://localhost/sample.pdf"
      >
        Download
      </a>
    </p>
  </object>
</div>
`;
